lib/whitehall_importer/create_migration.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "gds_api/whitehall_export"
+
+module WhitehallImporter
+  class CreateMigration
+    attr_reader :organisation_content_id, :document_type
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(organisation_content_id, document_type)
+      @organisation_content_id = organisation_content_id
+      @document_type = document_type
+    end
+
+    def call
+      whitehall_migration = ActiveRecord::Base.transaction do
+        record = WhitehallMigration.create!(organisation_content_id: organisation_content_id,
+                                            document_type: document_type)
+
+        whitehall_export = GdsApi.whitehall_export.document_list(organisation_content_id, document_type)
+
+        whitehall_export.each do |page|
+          page["documents"].each do |document|
+            WhitehallMigration::DocumentImport.create!(
+              whitehall_document_id: document["document_id"],
+              whitehall_migration_id: record.id,
+              state: "pending",
+            )
+          end
+        end
+        record
+      end
```

Comment:
For brevity we might want to break this method up now we have the luxury of this being in a dedicated class.

As a suggestion would be:

```
    def call
      whitehall_migration = ActiveRecord::Base.transaction do
        record = create_migration
        document_list.each { |page| create_document_imports(record, page) }
        record
      end
      
      whitehall_migration.document_imports.find_each do |document_import|
        WhitehallDocumentImportJob.perform_later(document_import)
      end
      
      whitehall_migration
    end
```

Reply:
Good idea 👍